packages/backend-data/src/convert_schema.ts

```diff
@@ -16,22 +29,51 @@ export const isModelSchema = (
   return (
     schema !== null &&
     typeof schema === 'object' &&
+    'transform' in schema &&
     typeof schema.transform === 'function'
   );
 };
 
+/**
+ * Determine if the input schema is a collection of model schemas, and perform type narrowing.
+ * @param schema the schema that might be a collection of model schemas
+ * @returns a boolean indicating whether the schema is a collection of derived model schema, with type narrowing
+ */
+export const isCombinedSchema = (
+  schema: DataSchemaInput
+): schema is DerivedCombinedSchema => {
+  return (
+    schema !== null &&
+    typeof schema === 'object' &&
+    'schemas' in schema &&
+    Array.isArray(schema.schemas)
+  );
+};
+
+// DO NOT EDIT THE FOLLOWING VALUES, UPDATES TO DB TYPE OR STRATEGY WILL RESULT IN DB REPROVISIONING
+const DYNAMO_DATA_SOURCE_STRATEGY = {
+  dbType: 'DYNAMODB',
+  provisionStrategy: 'AMPLIFY_TABLE',
+} as const;
+
+// Translate the external engine types to the config values
+const RDS_DB_TYPES = {
+  mysql: 'MYSQL',
+  postgresql: 'POSTGRES',
+} as const;
+
 /**
  * Given an input schema type, produce the relevant CDK Graphql Def interface
- * @param schema the input schema type
+ * @param schema TS schema builder definition or string GraphQL schema
+ * @param backendSecretResolver secret resolver
+ * @param stableBackendIdentifiers backend identifiers
  * @returns the cdk graphql definition interface
  */
 export const convertSchemaToCDK = (
-  schema: DataSchema
+  schema: DataSchema,
```

Comment:
I don't fully understand why these conversions are not part of the `data-construct` but `backend-data`

Reply:
IIRC this was an early decision due to certain JSII-imposed limitations in the data-construct

Reply:
Shouldn't this be then part of `aws-amplify/amplify-category-api` package and exported from there? the logic here is very core to the data construct and I'm not sure if the `backend-data` package is the right place for it. (e.g. what's the ownership for this code for maintenance purposes?)

Reply:
I'm not sure it belongs in the construct, but it might belong in the schema builder transform method?. It would be nice if all that the `backend-data` package had to do was call `.transform()` and any combining of schemas happened within that method. But I also don't understand that package well enough to make that call.

Ultimately, I'm not strictly opposed to this logic living here, but we should be careful about how it grows over time. Ideally this package is the thinnest possible translation layer between `defineData` and initializing the data construct. We should push as much logic as possible into either the schema builder or the construct itself.

Reply:
@iartemiev, can we try moving this logic to schema builder or the data-construct (or another package in data repo if there are JSII concerns)? I want to avoid situation where we keep adding and building more stuff here but the ownership remains elsewhere.